src/Analyzers/Core/CodeFixes/RemoveRedundantEqualityWithTrue/RemoveRedundantEqualityWithTrueCodeFixProvider.cs

```diff
@@ -0,0 +1,64 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Diagnostics.CodeAnalysis;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Editing;
+
+namespace Microsoft.CodeAnalysis.RemoveRedundantEqualityWithTrue
+{
+    [ExportCodeFixProvider(LanguageNames.CSharp, LanguageNames.VisualBasic), Shared]
+
+    internal sealed class RemoveRedundantEqualityWithTrueCodeFixProvider
+        : SyntaxEditorBasedCodeFixProvider
+    {
+        [ImportingConstructor]
+        [SuppressMessage("RoslynDiagnosticsReliability", "RS0033:Importing constructor should be [Obsolete]", Justification = "Used in test code: https://github.com/dotnet/roslyn/issues/42814")]
+        public RemoveRedundantEqualityWithTrueCodeFixProvider()
+        {
+        }
+
+        public override ImmutableArray<string> FixableDiagnosticIds => ImmutableArray.Create(IDEDiagnosticIds.RemoveRedundantEqualityWithTrueDiagnosticId);
+
+        internal override CodeFixCategory CodeFixCategory => CodeFixCategory.CodeStyle;
+
+        public override Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            foreach (var diagnostic in context.Diagnostics)
```

Comment:
i think the right pattern is to instead report this on the first diagnostic.  @mavasani  to confirm.

Reply:
@CyrusNajmabadi I believe that one of the analyzers I implemented in past was buggy and @mavasani suggested that I have the loop in RegisterCodeFixesAsync.
If the loop should be removed here, I hope you explain when it's needed and when it's not.

Reply:
i honestly don'tknow.  that' why i'd like @mavasani  to weigh in.  He knows best here and can explain fully :)

Reply:
ok.  givenhis comment, i'm ok with the loop here :)